app/models/resource.rb

```diff
@@ -25,13 +25,9 @@ def uses_onesky?
     onesky_project_id.present?
   end
 
-  def create_new_draft(language_id)
-    language = Language.find(language_id)
-
-    # TODO: disable this to prevent the API from overwriting existing translations within OneSky.
-    # TODO: This will probably need to be revisited -DF
-    # PageClient.new(self, language.code).push_new_onesky_translation
-    Translation.create!(resource: self, language: language)
+  def create_draft(language_id)
+    translation = Translation.latest_translation(id, language_id)
+    translation.nil? ? create_first_draft(language_id) : translation.create_new_version
```

Comment:
is the call to `.nil?` necessary?

Reply:
what about `translation&.create_new_version || create_first_draft(language_id)`?

Reply:
That feels dirty to me

Reply:
You're right. `translation ? translation.create_new_version : create_first_draft(language_id)` then